### DISCUSSION TITLE
Help with SSL Error

### DISCUSSION THREAD

**Discussion by wabash9000** [firebase/flutterfire#8473] [Q&A] [answered]
I don't think it is an issue with library and I'm sure I'm just doing something wrong, but can anyone help me figure out this SSL error.

I had everything working on my app and then when I went to test it for release it no longer worked.  I was using an older version of the FlutterFire libraries so I updated my version of flutter and installed the newest FlutterFire libraries and now I'm getting a bunch of errors. I worked through the errors I was having with the database, but now I'm failing to signInAnonymously.

E/flutter ( 8171): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: [firebase_auth/unknown] com.google.firebase.FirebaseException: An internal error has occurred. [ Read error:ssl=0x7c0cefbc40: Failure in SSL library, usually a protocol error
```
E/flutter ( 8171): error:100000d7:SSL routines:OPENSSL_internal:SSL_HANDSHAKE_FAILURE (external/boringssl/src/ssl/ssl_lib.cc:717 0x7c2693788f:0x00000000) ]
E/flutter ( 8171): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
E/flutter ( 8171): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:177:18)
E/flutter ( 8171): <asynchronous suspension>
E/flutter ( 8171): #2      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:377:43)
E/flutter ( 8171): <asynchronous suspension>
E/flutter ( 8171): #3      MethodChannelFirebaseAuth.signInAnonymously (package:firebase_auth_platform_interface/src/method_channel/method_channel_firebase_auth.dart:414:12)
E/flutter ( 8171): <asynchronous suspension>
E/flutter ( 8171): #4      FirebaseAuth.signInAnonymously (package:firebase_auth/src/firebase_auth.dart:447:35)
```

from my pubspec.yaml
firebase_auth: ^3.3.14
firebase_database: ^9.0.11
firebase_core: ^1.14.1

function calling signInAnonymously
```
  void load(VoidCallback callback) async {
    AppData.error("user.Load()");
    if (FirebaseAuth.instance.currentUser == null) {
      try {
        FirebaseAuth.instance.signInAnonymously().then((UserCredential result) {
          User user = result.user;
          _firebaseUser = user;
          AppData.error("Signed In Anonymously");
          loadUserData(callback);
        });
      } catch (exception) {
        _firebaseUser = null;
        AppData.error("Catch on FirebaseAuth.instance.signInAnonymously()");
        loadUserData(callback);
      }
    } else {
      AppData.error("LoadUserData for firebase User");
      loadUserData(callback);
    }
  }
```

Oddly that function has a try around the signInAnonymously, but it doesn't seem to be catching the SSL error.  AppData.error should write to the console when called. 

```

Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.10.4, on Microsoft Windows [Version 10.0.19044.1586], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.5)
[√] Android Studio (version 4.1)
[√] VS Code (version 1.66.2)
[√] Connected device (4 available)
[√] HTTP Host Availability
```

Test device is a Lenovo TB8504F Tab 8 running Android 8.1.0

Anyone have any ideas?

**Comment by wabash9000** [comment, marked as answer]
Let it sit over the weekend, figured it out first thing Monday morning.  Was on our corporate wifi but it didn't have internet or at least didn't have access to firebase. Threw it on wifi shared from my macbook and it worked perfectly. So if anyone is having this issue, check your internet connection.

**Reply by RamSakamuri** [reply]
@wabash9000    I am also facing same problem, when accessing firebase via unsecured wifi network e.g in public library
